packages/web3-utils/src/validation.ts

```diff
@@ -7,39 +9,37 @@ import {
 	InvalidNumberError,
 	NegativeIntegersInByteArrayError,
 	InvalidStringError,
+	InvalidBloomError,
+	InvalidBlockError,
+	InvalidTopicError,
+	InvalidCharCodeError,
+	InvalidAddressError,
 } from './errors';
 import { Bytes, HexString, Numbers } from './types';
 
-// TODO: implement later
-export const checkAddressChecksum = (_value: string): boolean => true;
-
-// TODO: Implement later
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const isAddress = (_value: string): boolean => {
-	// check if it has the basic requirements of an address
-	if (!/^(0x)?[0-9a-f]{40}$/i.test(_value)) {
-		return false;
-		// If it's ALL lowercase or ALL upppercase
-	}
-	if (/^(0x|0X)?[0-9a-f]{40}$/.test(_value) || /^(0x|0X)?[0-9A-F]{40}$/.test(_value)) {
-		return true;
-		// Otherwise check each case
-	}
-	return checkAddressChecksum(_value);
-};
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-export const isAddressv2 = (_value: string): boolean => true;
-
 export const isHexStrict = (hex: string) =>
 	typeof hex === 'string' && /^(-)?0x[0-9a-f]*$/i.test(hex);
 
+/**
+ * returns true if input is a hexstring, number or bigint
+ */
+export const isHex = (hex: Numbers): boolean =>
+	typeof hex === 'number' ||
+	typeof hex === 'bigint' ||
```

Comment:
Yea I agree bigint should return `false`.
From the post it seems like we can't check if the typeof `number` is prefixed with a `0x`. 
It seems like in `1.x` `isHex` returns true if the value is typeof `number` so it'll be a breaking change.
We can change it so it only accepts strings with a `0x` prefix, that'll have the exact same functionality as `isHexStrict` function

Reply:
I misread the post I linked, it would be possible if we did `.toString()`, but I think your point about it being a breaking change is the most important detail